generic_config_updater/patch_sorter.py

```diff
@@ -691,6 +855,63 @@ def _validate_table(self, table, config):
         # the only invalid case is if table exists and is empty
         return table not in config or config[table]
 
+class RequiredValueMoveValidator:
+    """
+    Check RequiredValueIdentifier class description first.
+
+    The validator checks the following:
+    - A move that is changing a requiring config, while the required path is not equal to the required value is rejected
+      E.g. A move that is changing "QUEUE" table while the corresponding "PORT" is not admin down is rejected
+    - A move that is changing the required path value to something other than the required value, while there are
+      requiring changes left is rejected
+      E.g. A move is changing "PORT" to admin up from down, while "QUEUE" table still have changes left is rejected.
+    """
+    def __init__(self, path_addressing):
+        self.path_addressing = path_addressing
+        self.identifier = RequiredValueIdentifier(path_addressing)
+
+    def validate(self, move, diff):
+        # ignore full config removal because it is not possible by JsonPatch lib
+        if move.op_type == OperationType.REMOVE and move.path == "":
+            return
+
+        current_config = diff.current_config
+        simulated_config = move.apply(current_config) # Config after applying just this move
+        target_config = diff.target_config # Final config after applying whole patch
+
+        # data dictionary:
+        # {
+        #   <path>: [(required_path, required_value), ...],
+        #   ...
+        # }
+        data = self.identifier.get_required_value_data([current_config, simulated_config, target_config])
+
+        # If move is changing a requiring path while the required path does not have the required value, reject the move
```

Comment:
>changing a requiring path while the required path [](http://example.com/codeflow?start=21&length=49)

I feel confused about 'requiring' vs 'required'. As your E.g., the move is changing a port-critical config which is not admin status, itself should not be requiring or required.

Reply:
The requiring config is the `port-critical` config can `QUEUE`, `BUFFER_PG`. The required config is the `PORT` config, and it is required to be down